front/components/agent_builder/triggers/WebhookEditionModal.tsx

```diff
@@ -0,0 +1,341 @@
+import { useUser } from "@app/lib/swr/user";
+import { useWebhookSourcesWithViews } from "@app/lib/swr/webhook_source";
+import type { LightWorkspaceType } from "@app/types";
+
+const webhookFormSchema = z.object({
+  name: z.string().min(1, "Name is required").max(255, "Name is too long"),
+  customPrompt: z.string(),
+  webhookSourceViewSId: z.string().min(1, "Select a webhook source"),
+  includePayload: z.boolean().default(false),
+});
+
+type WebhookFormData = z.infer<typeof webhookFormSchema>;
+
+interface WebhookEditionModalProps {
+  owner: LightWorkspaceType;
+  trigger?: AgentBuilderWebhookTriggerType;
+  isOpen: boolean;
+  onClose: () => void;
+  onSave: (trigger: AgentBuilderWebhookTriggerType) => void;
+}
+
+type WebhookOption = {
+  value: string;
+  label: string;
+  sourceName: string;
+  spaceName: string | null;
+};
+
+export function WebhookEditionModal({
+  owner,
+  trigger,
+  isOpen,
+  onClose,
+  onSave,
+}: WebhookEditionModalProps) {
+  const { user } = useUser();
+
+  const defaultValues = useMemo(
+    (): WebhookFormData => ({
+      name: "Webhook Trigger",
+      customPrompt: "",
+      webhookSourceViewSId: "",
+      includePayload: false,
+    }),
+    []
+  );
+
+  const form = useForm<WebhookFormData>({
+    resolver: zodResolver(webhookFormSchema),
+    defaultValues,
+  });
+
+  const selectedViewSId = form.watch("webhookSourceViewSId") ?? "";
+  const includePayload = form.watch("includePayload");
+
+  const { spaces } = useSpacesContext();
+  const { webhookSourcesWithViews, isWebhookSourcesWithViewsLoading } =
+    useWebhookSourcesWithViews({ owner });
+
+  // eslint-disable-next-line @typescript-eslint/prefer-nullish-coalescing
+  const isEditor = !trigger?.editor || trigger?.editor === user?.id;
+
+  const spaceById = useMemo(() => {
+    return new Map(spaces.map((space) => [space.sId, space.name]));
+  }, [spaces]);
+
+  const accessibleSpaceIds = useMemo(
+    () => new Set(spaceById.keys()),
+    [spaceById]
+  );
+
+  const webhookOptions = useMemo((): WebhookOption[] => {
+    const options: WebhookOption[] = [];
+
+    webhookSourcesWithViews.forEach((wsv) => {
+      wsv.views
+        .filter((view) => accessibleSpaceIds.has(view.spaceId))
+        .forEach((view) => {
+          options.push({
+            value: view.sId,
+            label: view.customName ?? wsv.name,
+            sourceName: wsv.name,
+            spaceName: spaceById.get(view.spaceId) ?? null,
+          });
+        });
+    });
+
+    return options;
+  }, [webhookSourcesWithViews, accessibleSpaceIds, spaceById]);
+
+  useEffect(() => {
+    if (!isOpen) {
+      form.reset(defaultValues);
+      return;
+    }
+
+    if (!trigger) {
+      form.reset(defaultValues);
+      return;
+    }
+
+    const includePayload =
+      trigger.configuration &&
+      "includePayload" in trigger.configuration &&
+      typeof trigger.configuration.includePayload === "boolean"
```

Comment:
Not sure if it is useful. Isn't it typed boolean already ?

Reply:
yep, good catch